cinder_plugin/tests/test_volume.py

```diff
@@ -339,4 +340,317 @@ def test_detach(self, wait_until_status_m, cinder_m, nova_m):
             status=volume.VOLUME_STATUS_AVAILABLE,
             num_tries=10,
             timeout=2,
+        )
+
+    def _simple_volume_ctx(self):
+        volume_id = '1234-5678'
+        volume_ctx = cfy_mocks.MockCloudifyContext(
+            node_id="node_id",
+            node_name="node_name",
+            properties={},
+            runtime_properties={
+                OPENSTACK_ID_PROPERTY: volume_id,
+            }
+        )
+        current_ctx.set(volume_ctx)
+        return volume_ctx, volume_id
+
+    @mock.patch('openstack_plugin_common.CinderClientWithSugar')
+    @mock.patch('time.sleep', mock.Mock())
+    def test_snapshot_create(self, cinder_m):
+        cinder_instance = cinder_m.return_value
+        volume_ctx, volume_id = self._simple_volume_ctx()
+
+        # Snapshot
+        cinder_instance.backups.create = mock.Mock()
+        cinder_instance.volume_snapshots.create = mock.Mock()
+
+        volume.snapshot_create(ctx=volume_ctx, snapshot_name="snapshot_name",
+                               snapshot_incremental=True,
+                               snapshot_type="abc")
+
+        cinder_instance.backups.create.assert_not_called()
+        cinder_instance.volume_snapshots.create.assert_called_once_with(
+            '1234-5678', description='abc',
+            force=True, metadata=None,
+            name='vol-1234-5678-snapshot_name')
+
+        # Backup
+        cinder_instance.backups.create = mock.Mock()
+        cinder_instance.volume_snapshots.create = mock.Mock()
+
+        volume.snapshot_create(ctx=volume_ctx, snapshot_name="backup_name",
+                               snapshot_incremental=False)
+
+        cinder_instance.backups.create.assert_called_once_with(
+            '1234-5678',
+            name='vol-1234-5678-backup_name')
+        cinder_instance.volume_snapshots.create.assert_not_called()
+
+    @mock.patch('openstack_plugin_common.CinderClientWithSugar')
+    @mock.patch('time.sleep', mock.Mock())
```

Comment:
@0lvin same here for time , this should be remobved

Reply:
@mabuaisha code removes volume and for case "fast" remove recheck that have deleted snapshot after small sleep. And only after that raise exception.